test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,978 @@
+        }
+
+        Project.init({
+          name: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            defaultValue: 'unknown',
+            validate: {
+              isIn: [['unknown', 'hello', 'test']],
+            },
+          },
+        }, { sequelize });
+
+        return { Project };
+      });
+
+      it('correctly throws an error using create method ', async () => {
+        await vars.Project.create({})
+          .then(() => {
+            throw new Error('Validation must be failed');
+          }, async () => {
+            // fail is ok
+          });
+      });
+
+      it('correctly throws an error using create method with default generated messages', async () => {
+        try {
+          await vars.Project.create({});
+        } catch (error) {
+          if (error instanceof ValidationError) {
+            expect(error).to.have.property('name', 'SequelizeValidationError');
+            expect(error.message).equal('notNull violation: Project.name cannot be null');
+            expect(error.errors).to.be.an('array').and.have.length(1);
+            expect(error.errors[0]).to.have.property('message', 'Project.name cannot be null');
+          }
+        }
+      });
+    });
+  });
+
+  it('correctly validates using custom validation methods', async () => {
+    class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+      declare name: string;
+    }
+
+    User.init({
+      name: {
+        type: DataTypes.STRING,
+        validate: {
+          customFn(val: string, next: (arg0?: string | undefined) => void) {
+            if (val !== '2') {
+              next('name should equal \'2\'');
+            } else {
+              next();
+            }
+          },
+        },
+      },
+    }, { sequelize });
+
+    const failingUser = User.build({ name: '3' });
+
+    const error = await expect(failingUser.validate()).to.be.rejected;
+
+    expect(error).to.be.an.instanceOf(Error);
+
+    if (error instanceof ValidationError) {
+      expect(error.get('name')[0].message).to.equal('name should equal \'2\'');
+    }
+
+    const successfulUser = User.build({ name: '2' });
+
+    await expect(successfulUser.validate()).not.to.be.rejected;
+  });
+
+  it('supports promises with custom validation methods', async () => {
+    class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+      declare name: string;
+    }
+
+    User.init({
+      name: {
+        type: DataTypes.STRING,
+        validate: {
+          async customFn(val: string) {
+            await User.findAll();
+            if (val === 'error') {
+              throw new Error('Invalid username');
+            }
+          },
+        },
+      },
+    }, { sequelize });
+
+    await User.sync();
+    const error = await expect(User.build({ name: 'error' }).validate()).to.be.rejected;
+    expect(error).to.be.instanceof(ValidationError);
```

Comment:
use assert instead of expect for this so TypeScript knows it's that type

Reply:
assert works good!